implementations/rust/ockam/ockam_identity/src/authority.rs

```diff
@@ -0,0 +1,234 @@
+use crate::alloc::string::ToString;
+use ockam_core::compat::boxed::Box;
+use ockam_core::compat::collections::HashMap;
+use ockam_core::compat::string::String;
+use ockam_core::errcode::{Kind, Origin};
+use ockam_core::{AsyncTryClone, Error, Message, Result, Route, Routed, Worker};
+
+use crate::authenticated_storage::mem::InMemoryStorage;
+use ockam_node::Context;
+use ockam_vault::Vault;
+use AuthorityRequest::*;
+use AuthorityResponse::*;
+
+use crate::credential::Credential;
+use crate::{Identity, IdentityIdentifier, PublicIdentity};
+use anyhow::anyhow;
+use serde::{Deserialize, Serialize};
+
+/// This struct provides a simplified Authority which can be used in test scenarios
+/// by starting it as a Worker on any given node.
+///
+/// Note that the storage associated to the Authority identity will not persist between runs.
+pub struct Authority {
+    identity: Identity<Vault, InMemoryStorage>,
+}
+
+/// This trait provides an interface for an Authority so that it can be called directly
+/// or via a worker by sending messages
+#[ockam_core::async_trait]
+pub trait AuthorityApi {
+    /// Return the Authority public identity
+    async fn public_identity(&self) -> Result<PublicIdentity>;
+
+    /// Create an authenticated credential for an attribute name/value pair
+    /// and a given subject
+    async fn get_attribute_credential(
+        &self,
+        subject: &IdentityIdentifier,
+        attribute_name: &str,
+        attribute_value: &str,
+    ) -> Result<Credential>;
+
+    /// Create an authenticated credential for a list of attribute name/value pairs
+    /// and a given subject
+    async fn get_credential(
+        &self,
+        subject: &IdentityIdentifier,
+        attributes: HashMap<String, String>,
+    ) -> Result<Credential>;
+}
+
+impl Authority {
```

Comment:
if this is going to be used on tests only,   would it be possible to move out of here and into something only available on tests?.
If not,    I think we should rename it to something else,   like   `ToyCredentialIssuer` or something.

Reply:
I think that it's on the same level as an `InMemoryVault`. My inclination is that, at this stage it is more valuable to leave it in the toolkit in order to let people build prototypes more easily rather than enforcing a strong separation between "production" code and "test/prototype" code. 

That being said we could maybe get a better control on the situation by putting this behind a feature flag. What do you think?

Reply:
I think that `CredentialIssuer` might also be a better name.

Reply:
Every identity should be able to issue credentials.

CredentialIssuer is a better name, Authority is implied trust in some context.
Any identity should be able to issue credentials but it may not be an authority in any context.

Reply:
I changed the name